src/ec2_metadata.py

```diff
@@ -24,8 +27,26 @@ def __init__(self, session=None):
         if session is None:
             session = requests.Session()
         self._session = session
+        self._token_updated_at = 0
+
+    def _ensure_fresh_token(self):
+        """ Update the metadata token if needed.
+
+        Tokens are rotated 1 minute before they would expire.
+        """
+        now = time.time()
+        if now - self._token_updated_at > (TOKEN_TTL_SECONDS - 60):
+            token = self._session.put(
+                "http://169.254.169.254/latest/api/token",
```

Comment:
Don't use `latest`, which is a moving target. Use and update the hardcoded URL at the top of the file - which needs bumping to use the version 2 URL's, since none of the other functions will use v2 until that's done.